CMakeLists.txt

```diff
@@ -513,11 +513,22 @@ OCV_OPTION(OPENCV_ENABLE_MEMALIGN     "Enable posix_memalign or memalign usage"
 
 OCV_OPTION(ENABLE_PYLINT              "Add target with Pylint checks"                            (BUILD_DOCS OR BUILD_EXAMPLES) IF (NOT CMAKE_CROSSCOMPILING AND NOT APPLE_FRAMEWORK) )
 OCV_OPTION(ENABLE_FLAKE8              "Add target with Python flake8 checker"                    (BUILD_DOCS OR BUILD_EXAMPLES) IF (NOT CMAKE_CROSSCOMPILING AND NOT APPLE_FRAMEWORK) )
+OCV_OPTION(CV_HAVE_FILESYSTEM_SUPPORT "Enable filesystem support" NOT EMSCRIPTEN AND NOT WINRT AND NOT WINCE)
```

Comment:
To avoid questions about used condition it make sense to use this form:

```
OCV_OPTION(OPENCV_DISABLE_FILESYSTEM_SUPPORT "Disable filesystem support" OFF)
```

which adds definition `OPENCV_HAVE_FILESYSTEM_SUPPORT=0` (see related comment)

(This approach is similar to disabled threading support PR)